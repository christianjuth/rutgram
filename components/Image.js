import React, { Fragment } from 'react';
import { Image } from 'react-native';

function StyledImage(props) {
  return(
    <Fragment>
      <Image
        {...props}
        style={{width: '100%', paddingTop: '100%', position: 'absolute'}}
        resizeMode='cover'
        blurRadius={30}
      />
      <Image
        {...props}
        style={{width: '100%', paddingTop: '100%'}}
        resizeMode='contain'
      />
    </Fragment>
  );
}

export default StyledImage;
