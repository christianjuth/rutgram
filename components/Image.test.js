import React from 'react';
import { Image as RNImage } from 'react-native';
import renderer from 'react-test-renderer';
import Image from './Image';

describe('<Image />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Image srouce={{uri: ''}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('two images', () => {
    const tree = renderer.create(<Image/>);
    let images = tree.root.findAllByType(RNImage);
    expect(images.length).toBe(2);
  });
});
