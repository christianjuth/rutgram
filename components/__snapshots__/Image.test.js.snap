// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image /> renders correctly 1`] = `
Array [
  <Image
    blurRadius={30}
    resizeMode="cover"
    srouce={
      Object {
        "uri": "",
      }
    }
    style={
      Object {
        "paddingTop": "100%",
        "position": "absolute",
        "width": "100%",
      }
    }
  />,
  <Image
    resizeMode="contain"
    srouce={
      Object {
        "uri": "",
      }
    }
    style={
      Object {
        "paddingTop": "100%",
        "width": "100%",
      }
    }
  />,
]
`;
